scripts/check_for_inconsistent_pandas_namespace.py

```diff
@@ -15,43 +15,50 @@
 though note that you may need to manually fixup some imports and that you will also
 need the additional dependency `tokenize-rt` (which is left out from the pre-commit
 hook so that it uses the same virtualenv as the other local ones).
+
+The general structure is similar to that of some plugins from
+https://github.com/asottile/pyupgrade .
 """
 
 import argparse
 import ast
+import sys
 from typing import (
     MutableMapping,
+    NamedTuple,
     Optional,
     Sequence,
     Set,
-    Tuple,
 )
 
-ERROR_MESSAGE = "Found both `pd.{name}` and `{name}` in {path}"
-EXCLUDE = {
-    "eval",  # built-in, different from `pd.eval`
-    "np",  # pd.np is deprecated but still tested
-}
-Offset = Tuple[int, int]
+ERROR_MESSAGE = (
+    "{path}:{lineno}:{col_offset}: "
+    "Found both '{prefix}.{name}' and '{name}' in {path}"
+)
+
+
+class OffsetWithNamespace(NamedTuple):
+    lineno: int
+    col_offset: int
+    namespace: str
 
 
 class Visitor(ast.NodeVisitor):
     def __init__(self) -> None:
-        self.pandas_namespace: MutableMapping[Offset, str] = {}
-        self.no_namespace: Set[str] = set()
+        self.pandas_namespace: MutableMapping[OffsetWithNamespace, str] = {}
+        self.imported_from_pandas: Set[str] = set()
```

Comment:
Outside scope of the PR of course, but curious how hard it would be to generalize this to _any_ namespace? Then you could see it being a helpful linting check even outside of pandas.

Reply:
That's a great idea, thanks!